portal.go

```diff
@@ -963,7 +963,14 @@ func (portal *Portal) storeReactionInDB(
 
 func (portal *Portal) addSignalQuote(content *event.MessageEventContent, quote *signalmeow.IncomingSignalMessageQuoteData) {
 	if quote != nil {
-		originalMessage := portal.bridge.DB.Message.FindBySenderAndTimestamp(quote.QuotedSender, quote.QuotedTimestamp)
+		puppet := portal.bridge.DB.Puppet.GetBySignalID(quote.QuotedSender)
+		if puppet == nil {
+			portal.log.Warn().Msgf("Couldn't find puppet for quoted sender %s", quote.QuotedSender)
+			return
+		}
+		originalMessage := portal.bridge.DB.Message.GetBySignalID(
+			puppet.SignalID, quote.QuotedTimestamp, portal.ChatID, portal.Receiver,
```

Comment:
Shouldn't this work just by using `quote.QuotedSender` instead of `puppet.SignalID` and not fetching the puppet at all?

Reply:
Yes, but [the previous version of the bridge did this](https://github.com/mautrix/signal/blob/master/mautrix_signal/portal.py#L1147), so I did it here too.  I presume the benefit of the puppet lookup is to ensure the validity of the bridged user who sent the message that the quote refers to.

Reply:
I think the previous bridge just did that to deal with the phone number -> uuid migration. There's no need to ensure the validity, if the message row exists it's valid enough.